TmdbList addition

KellyMahan opened the pull request:
Added new class TmdbList with 4 new methods to retrieve data from tmdb.

list = TmdbList.upcoming
list = TmdbList.now_playing
list = TmdbList.top_rated
list = TmdbList.popular

each method has an optional page and language arguments
list = TmdbList.popular(2,"EN")

a TmdbList instance has theses accessors

attr_accessor :page, :movie_ids, :total_pages, :total_results, :movies_data, :movies

list.movies_data is the original json data
list.movies is an array of TmdbList::Movie objects.

a TmdbList::Movie object has these accessors.
attr_accessor :backdrop_path, :id, :original_title, :release_date, :poster_path, :title, :vote_average, :vote_count

KellyMahan committed:
wrote TmdbUpcoming class and methods with fixture and test

KellyMahan committed:
need to set uri.query_values for live data

KellyMahan committed:
changed TmdbUpcoming to more generic TmdbList. Also added a subclass to handle individual movie info from the results.

Irio commented:
Sorry for the delay, @KellyMahan, I was fixing the test suite inside TravisCI before read and test your pull request.

Great addition in the gem. Could you document `TmdbList` methods in README file?

KellyMahan committed:
updated readme and made NilData class name consistent with others

KellyMahan committed:
readme formatting

KellyMahan committed:
added a new method to the TmdbList class for retrieving which movies were changed.

KellyMahan committed:
changes for lists to let movie_ids pull from the movies array every time instead of being set from the initialization.

KellyMahan committed:
forgot to add file

KellyMahan committed:
added a rate limit time option so the user isn't required to handle rate limit issues.

KellyMahan committed:
Didn't think that one through.

KellyMahan commented:
Sorry for all the commits. Were will be using this gem in production soon and I've been adding to it as needed.

KellyMahan committed:
setting rate_limit to 0 for tests, otherwise tests take over 100 seconds to run

Irio commented:
@KellyMahan Can you verify your last commits? It seems your last commit introduced a conflict, so I can't merge it.

You can also see [Travis CI](https://travis-ci.org/Irio/ruby-tmdb/pull_requests) build history.

KellyMahan committed:
Merge branch 'master' of git://github.com/Irio/ruby-tmdb

Conflicts:
	test/unit/test_direct_require.rb

KellyMahan committed:
Provided options were being overridden. This fixes that.

KellyMahan committed:
even though results were returned using the query param didn't work the way I expected it. This creates the proper url for tmdb

KellyMahan committed:
added ability to rate movies. added tests, readme for rating movies
Tmdb.base_api_url can now be set so apiary private proxy logging can be used

Irio commented on lib/ruby-tmdb3/tmdb_list.rb:
```diff
@@ -0,0 +1,75 @@
+    options = {
+      page: 1,
+      start_date: Time.now.strftime("%Y-%m-%d"),
+      end_date: (Time.now+60*60*24).strftime("%Y-%m-%d"),
```
Can you use `Time.now + 86400` and add a comment to show why this number?

KellyMahan replied:
done

Irio commented on lib/ruby-tmdb3/tmdb_movie.rb:
```diff
@@ -1,5 +1,19 @@
+      session_id = response["guest_session_id"]
+    end
+    response = Tmdb.api_call("movie/rating", {id: id, guest_session_id: session_id, value: rating}, options[:language], true)
+    return response["status_code"], session_id, response["status_message"]
```
How you feel returning a Hash with this variables? At least for me, an Array doesn't explains much about its values.

KellyMahan replied:
added in new commits

Irio commented on lib/ruby-tmdb3/tmdb_list.rb:
```diff
@@ -0,0 +1,75 @@
+      page: 1,
+      start_date: Time.now.strftime("%Y-%m-%d"),
+      end_date: (Time.now+60*60*24).strftime("%Y-%m-%d"),
+      language: "EN"
```
Do you have a reason to not use `Tmdb.default_language`?

KellyMahan replied:
language is ignored for the changes method so I removed it.

KellyMahan committed:
Merge branch 'master' of git://github.com/Irio/ruby-tmdb

KellyMahan committed:
added comment about end time.
removed language option from the changes method. It isn't used.
returning hash of values when rating a movie instead of an array

KellyMahan committed:
modified tests for new rating response

KellyMahan committed:
1.8.7 syntax for hash

KellyMahan committed:
more 1.8.7 syntax compatible changes

KellyMahan committed:
missed another 1.8.7 syntax change

KellyMahan committed:
missed another 1.8.7

KellyMahan committed:
Merge branch 'master' of git://github.com/Irio/ruby-tmdb

KellyMahan committed:
changes for cast images

KellyMahan committed:
wrong order of url stub caused an issue

KellyMahan committed:
fix for popular list

KellyMahan committed:
better debugging

KellyMahan committed:
added Kelly Mahan to authors

KellyMahan committed:
added an error check when a list returns a null movie (I think this is a bug in the api)

KellyMahan committed:
updated version

vgkids commented:
It looks like TmdbList never made it into Master? Any chance that could get merged?

Irio commented:
Sorry, @vgkids. I can't merge it while the branch have conflicts.

@KellyMahan do you still want to solve them?

KellyMahan committed:
updated base url

KellyMahan committed:
updated version

KellyMahan committed:
missed version in gemspec
